db/api/v1/routes/validate.js

```diff
@@ -14,37 +14,34 @@ const utils = require('../utils');
  * @param {*} parser - callback that parses the argument to a value or returns a parse failure error
  * @returns Express middleware
  */
-const validate = (extractParam, paramName, parser) =>
+const validate = (extractParam, paramName, parser, required = false) =>
     (req, res, next) => {
         const arg = extractParam(req, paramName);
 
-        if (arg !== undefined && typeof arg !== 'string') {
-            throw new Error(`validate: Parameter extracter returned a value with type '${typeof arg}'. Parameter extracters may only return values with the type 'string'.`);
-        }
-
-        if (!req.valid) {
-            req.valid = {};
-        }
+        if (req.valid) throw new Error('The request parameter "valid" is reserved for validation middleware. Please change your request.');
```

Comment:
This breaks the test 'does not delete any existing req.valid entries' test in validate.test.js. I think it would be best to lock down not being able to use 'valid' in requests

Reply:
I decided to skip the tests